raiden/network/rpc/client.py

```diff
@@ -312,6 +381,743 @@ def estimate_and_transact(classobject, callobj, *args):
+                'transaction_hash length must be 32 (it might be hex encode)'
+            )
+
+        transaction_hash = data_encoder(transaction_hash)
+        return self.call('eth_getTransactionReceipt', transaction_hash)
+
+    def eth_getCode(self, address, block='latest'):
+        """ Returns code at a given address.
+
+        Args:
+            address: An address.
+            block_number: Integer block number, or the string "latest",
+                "earliest" or "pending".
+        """
+        if address.startswith('0x'):
+            warnings.warn(
+                'address seems to be already encoded, this will result '
+                'in unexpected behavior'
+            )
+
+        if len(address) != 20:
+            raise ValueError(
+                'address length must be 20 (it might be hex encode)'
+            )
+
+        return self.call(
+            'eth_getCode',
+            address_encoder(address),
+            block,
+        )
+
+    def eth_getTransactionByHash(self, transaction_hash):
+        """ Returns the information about a transaction requested by
+        transaction hash.
+        """
+
+        if transaction_hash.startswith('0x'):
+            warnings.warn(
+                'transaction_hash seems to be already encoded, this will'
+                ' result in unexpected behavior'
+            )
+
+        if len(transaction_hash) != 32:
+            raise ValueError(
+                'transaction_hash length must be 32 (it might be hex encode)'
+            )
+
+        transaction_hash = data_encoder(transaction_hash)
+        return self.call('eth_getTransactionByHash', transaction_hash)
+
+    def poll(self, transaction_hash, confirmations=None, timeout=None):
+        """ Wait until the `transaction_hash` is applied or rejected.
+        If timeout is None, this could wait indefinitely!
+
+        Args:
+            transaction_hash (hash): Transaction hash that we are waiting for.
+            confirmations (int): Number of block confirmations that we will
+                wait for.
+            timeout (float): Timeout in seconds, raise an Excpetion on
+                timeout.
+        """
+        if transaction_hash.startswith('0x'):
+            warnings.warn(
+                'transaction_hash seems to be already encoded, this will'
+                ' result in unexpected behavior'
+            )
+
+        if len(transaction_hash) != 32:
+            raise ValueError(
+                'transaction_hash length must be 32 (it might be hex encode)'
+            )
+
+        transaction_hash = data_encoder(transaction_hash)
+
+        deadline = None
+        if timeout:
+            deadline = gevent.Timeout(timeout)
+            deadline.start()
+
+        try:
+            # used to check if the transaction was removed, this could happen
+            # if gas price is to low:
+            #
+            # > Transaction (acbca3d6) below gas price (tx=1 Wei ask=18
```

Comment:
This comment here seems totally random. Perhaps remove it?

Reply:
It's the error message reported by geth (or it's a few version back)